eva-accession-clustering/src/main/java/uk/ac/ebi/eva/accession/clustering/configuration/batch/io/RSMergeAndSplitCandidatesReaderConfiguration.java

```diff
@@ -48,53 +49,53 @@ public class RSMergeAndSplitCandidatesReaderConfiguration {
 
     public static final String ASSEMBLY_FIELD = "inactiveObjects.seq";
 
-    private static final String SORT_FIELD = "accession";
-
     public static final String EVENT_TYPE_FIELD = "eventType";
 
     public static final EventType SPLIT_CANDIDATES_EVENT_TYPE = EventType.RS_SPLIT_CANDIDATES;
 
     public static final EventType MERGE_CANDIDATES_EVENT_TYPE = EventType.RS_MERGE_CANDIDATES;
 
-    public static Query getSplitCandidatesQuery(String assemblyAccession) {
-        return Query.query(where(ASSEMBLY_FIELD).is(assemblyAccession))
-                    .addCriteria(where(EVENT_TYPE_FIELD).is(SPLIT_CANDIDATES_EVENT_TYPE.toString()));
+    public static Criteria getSplitCandidatesCriteria(String assemblyAccession) {
+        return where(DBCollection.ID_FIELD_NAME).regex("^RSSC_" + assemblyAccession + "_.*");
     }
 
-    public static Query getMergeCandidatesQuery(String assemblyAccession) {
-        return Query.query(where(ASSEMBLY_FIELD).is(assemblyAccession))
-                    .addCriteria(where(EVENT_TYPE_FIELD).is(MERGE_CANDIDATES_EVENT_TYPE.toString()));
+    public static Criteria getMergeCandidatesCriteria(String assemblyAccession) {
+        return where(DBCollection.ID_FIELD_NAME).regex("^RSMC_" + assemblyAccession + "_.*");
     }
 
     @Bean(RS_SPLIT_CANDIDATES_READER)
-    public ItemReader<SubmittedVariantOperationEntity> rsSplitCandidatesReader(MongoTemplate mongoTemplate,
-                                                                               InputParameters parameters) {
-        MongoItemReader<SubmittedVariantOperationEntity> mongoItemReader = new MongoItemReader<>();
+    public MongoDbCursorItemReader<SubmittedVariantOperationEntity> rsSplitCandidatesReader(MongoTemplate mongoTemplate,
+                                                                                            InputParameters parameters)
+    {
+        MongoDbCursorItemReader<SubmittedVariantOperationEntity> mongoItemReader = new MongoDbCursorItemReader<>();
         mongoItemReader.setTemplate(mongoTemplate);
         mongoItemReader.setTargetType(SubmittedVariantOperationEntity.class);
         mongoItemReader.setCollection(SUBMITTED_VARIANT_OPERATIONS_COLLECTION);
 
         //See https://docs.google.com/spreadsheets/d/1KQLVCUy-vqXKgkCDt2czX6kuMfsjfCc9uBsS19MZ6dY/#rangeid=1213746442
-        Query query = getSplitCandidatesQuery(parameters.getAssemblyAccession());
+        Meta meta = new Meta();
+        meta.addFlag(Meta.CursorOption.NO_TIMEOUT);
+        Query query = query(getSplitCandidatesCriteria(parameters.getAssemblyAccession()));
+        query.setMeta(meta);
         mongoItemReader.setQuery(query);
-        mongoItemReader.setPageSize(parameters.getChunkSize());
-        mongoItemReader.setSort(Collections.singletonMap(SORT_FIELD, Sort.Direction.ASC));
```

Comment:
Initially I had a misguided opinion that having a sorted reader would enable us to resume merge and splits from where we left off. But sorting seems to take a big toll on performance for queries with conditions (as opposed to full collection scans). Since we are not doing resumption at the moment anyway, this seems to impose an extra performance penalty which is not worthwhile.